File: app/src/main/java/com/statuses/statussavers/Splash.java
package com.statuses.statussavers;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import android.Manifest;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.os.Handler;
import android.view.WindowManager;

import static android.os.Build.VERSION.SDK_INT;

public class Splash extends AppCompatActivity {
    final static int SPLASH_TIMER = 2000;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        ActionBar actionBar = this.getSupportActionBar();
        if (actionBar != null) {
            actionBar.hide();
        }
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                Intent intent;
                if (checkBothPermission()) {
                    intent = new Intent(Splash.this, MainActivity.class);
                } else {
                     intent = new Intent(Splash.this, StartActivity.class);
                }
                startActivity(intent);
                finish();
            }
        }, SPLASH_TIMER);

    }

    private Boolean checkBothPermission() {
        Boolean readwrite = false;
        Boolean path = false;
        if(SDK_INT >=30) {
            path = readDataFromPrefs();
            readwrite = readPermission();
        }
        else {
            path = true;
            readwrite = readPermission();

        }
        return path && readwrite;
    }

    private Boolean readPermission() {
        if (SDK_INT <=29) {
            if (ContextCompat.checkSelfPermission(Splash.this, Manifest.permission.WRITE_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED) {
                return true;
            }
            else {
                return false;
            }
        } else {
            SharedPreferences sh = Splash.this.getSharedPreferences("PERMISSION", Context.MODE_PRIVATE);
            String uri;
            if (SDK_INT >= 33) {
                uri = sh.getString("readwrite33", "");
            } else {
                uri = sh.getString("readwrite", "");
            }
            if (uri != null) {
                if (uri.isEmpty()) {
                    return false;
                }
            }
            return true;
        }
    }

    private Boolean readDataFromPrefs() {
        SharedPreferences sh = Splash.this.getSharedPreferences("DATA_PATH", Context.MODE_PRIVATE);
        String uri = sh.getString("PATH", "");
        if(uri!=null) {
            if (uri.isEmpty()) {
                return false;
            }
        }
        return true;
    }
}

File: app/src/main/java/com/statuses/statussavers/Splash.kt
package com.statuses.statussavers

import android.Manifest
import android.content.Intent
import android.content.pm.PackageManager
import android.os.Build.VERSION
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.view.WindowManager
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat

class Splash : AppCompatActivity() {
    private var subscriptionManager: SubscriptionManager? = SubscriptionManager()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        val actionBar = this.supportActionBar
        actionBar?.hide()
        window.setFlags(
            WindowManager.LayoutParams.FLAG_FULLSCREEN,
            WindowManager.LayoutParams.FLAG_FULLSCREEN
        )

        Handler().postDelayed({
            initialChecks()
        }, SPLASH_TIMER.toLong())
    }

    private fun initialChecks() {
        subscriptionManager?.initBillingClient(this) { isSetupFinished ->
            if (isSetupFinished) {
                subscriptionManager?.checkSubscription { isSubscribed ->
                    Handler(Looper.getMainLooper()).post {
                        HelperClass.adsDisabled = isSubscribed
                        startPremissionCheck()
                    }
                }
            } else {
                Handler(Looper.getMainLooper()).post {
                    startPremissionCheck()
                }
            }
        }
    }

    private fun startPremissionCheck() {
            val intent = if (checkBothPermission()) {
                Intent(this@Splash, MainActivity::class.java)
            } else {
                Intent(this@Splash, StartActivity::class.java)
            }
            startActivity(intent)
            finish()
    }

    private fun checkBothPermission(): Boolean {
        var readwrite = false
        var path = false
        if (VERSION.SDK_INT >= 30) {
            path = readDataFromPrefs()
            readwrite = readPermission()
        } else {
            path = true
            readwrite = readPermission()
        }
        return path && readwrite
    }

    private fun readPermission(): Boolean {
        if (VERSION.SDK_INT <= 29) {
            return if (ContextCompat.checkSelfPermission(
                    this@Splash,
                    Manifest.permission.WRITE_EXTERNAL_STORAGE
                ) == PackageManager.PERMISSION_GRANTED
            ) {
                true
            } else {
                false
            }
        } else {
            val sh = this@Splash.getSharedPreferences("PERMISSION", MODE_PRIVATE)
            val uri = if (VERSION.SDK_INT >= 33) {
                sh.getString("readwrite33", "")
            } else {
                sh.getString("readwrite", "")
            }
            if (uri != null) {
                if (uri.isEmpty()) {
                    return false
                }
            }
            return true
        }
    }

    private fun readDataFromPrefs(): Boolean {
        val sh = this@Splash.getSharedPreferences("DATA_PATH", MODE_PRIVATE)
        val uri = sh.getString("PATH", "")
        if (uri != null) {
            if (uri.isEmpty()) {
                return false
            }
        }
        return true
    }

    companion object {
        const val SPLASH_TIMER: Int = 1000
    }

    override fun onDestroy() {
        super.onDestroy()
        subscriptionManager?.endConnection()
        subscriptionManager = null
    }
}